src/App.jsx

```diff
@@ -26,75 +29,122 @@ class App extends React.Component {
     };
   }
 
-setKeywords1 = (keywords1) => this.setState({ keywords1 })
-setKeywords2 = (keywords2) => this.setState({ keywords2 })
+  componentDidMount() {
+    this.getData();
+  }
 
-manageError = () => {
-  this.setState({ errorShow : !this.state.errorShow });
-}
+  setKeywords1 = (keywords1) => this.setState({ keywords1 });
+  setKeywords2 = (keywords2) => this.setState({ keywords2 });
 
-searchIngredient1 = () => {
-  
-  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords1 !== undefined ? this.state.keywords1 : ''}`)
-    .then((response) => response.data)
-    .then((data) => {
-      this.setState({ cocktails1: data.drinks });
-    });
-}
+  manageError = () => {
+    this.setState({ errorShow : !this.state.errorShow });
+  }
 
-searchIngredient2 = () => {
+  searchIngredient1 = () => {
   
-  Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords2 !== undefined ? this.state.keywords2 : ''} `)
-    .then((response) => response.data)
-    .then((data) => {
-      this.setState({ cocktails2: data.drinks });
-    });
-}
+    Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords1 !== undefined ? this.state.keywords1 : ''}`)
+      .then((response) => response.data)
+      .then((data) => {
+        this.setState({ cocktails1: data.drinks });
+      });
+  }
+  
+  searchIngredient2 = () => {
+    
+    Axios.get(`https://www.thecocktaildb.com/api/json/v1/1/filter.php?i=${this.state.keywords2 !== undefined ? this.state.keywords2 : ''} `)
+      .then((response) => response.data)
+      .then((data) => {
+        this.setState({ cocktails2: data.drinks });
+      });
+  }
 
-compare = () => {
-  if (this.state.cocktails1 === undefined || this.state.cocktails2 === undefined) {
-    this.setState ({ errorMessage : 'Il manque un ingrédient' })
-    this.setState ({ errorShow : true })
-  } else {
-    this.setState ({ intersection: [] });
-    this.setState ({ intersection:  this.state.cocktails1.filter((cocktail1) => this.state.cocktails2.findIndex(cocktail2 => cocktail1.strDrink === cocktail2.strDrink) !== -1)})
-    if (this.state.intersection.length === 0) {
-      this.setState ({ errorMessage :'aucun cocktail ne contient ces ingrédients'});
+  compare = () => {
+    if (this.state.cocktails1 === undefined || this.state.cocktails2 === undefined) {
+      this.setState ({ errorMessage : 'Il manque un ingrédient' })
       this.setState ({ errorShow : true })
-
+    } else {
+      this.setState ({ intersection: [] });
+      this.setState ({ intersection:  this.state.cocktails1.filter((cocktail1) => this.state.cocktails2.findIndex(cocktail2 => cocktail1.strDrink === cocktail2.strDrink) !== -1)})
+      if (this.state.intersection.length === 0) {
+        this.setState ({ errorMessage :'aucun cocktail ne contient ces ingrédients'});
+        this.setState ({ errorShow : true })
+  
+      }
```

Comment:
this may use the previous value of the state: make a variable to store the intersection, juste before the line 66, and use it in the setState line 67 and in the if line 68